x/incentives/keeper/gauge.go

```diff
@@ -278,3 +286,27 @@ func (k Keeper) GetEpochInfo(ctx sdk.Context) epochtypes.EpochInfo {
 	params := k.GetParams(ctx)
 	return k.ek.GetEpochInfo(ctx, params.DistrEpochIdentifier)
 }
+
+// chargeFeeIfSufficientFeeDenomBalance charges fee in the base tx fee denom on the address if the address has
+// balance that is less than fee + amount of the coin from gaugeCoins that is of base tx fee denom.
+// gaugeCoins might not have a coin of tx base fee denom. In that case, only fee is compared to balance.
+// Returns nil on success, error otherwise.
+func (k Keeper) chargeFeeIfSufficientFeeDenomBalance(ctx sdk.Context, address sdk.AccAddress, fee int64, gaugeCoins sdk.Coins) (err error) {
+	// Send creation fee to community pool
+	feeDenom, err := k.tk.GetBaseDenom(ctx)
+	if err != nil {
+		return err
+	}
```

Comment:
I think we can just hardcode this to "uosmo" for now, instead of needing to query feeDenom.

Or if you want, you can use `appparams.BaseCoinUnit`